lib/db.c

```diff
@@ -1271,10 +1271,13 @@ grn_rc
 grn_db_check_name(grn_ctx *ctx, const char *name, unsigned int name_size)
 {
   int len;
-  const char *name_end = name + name_size;
-  if (name_size > 0 && *name == GRN_DB_PSEUDO_COLUMN_PREFIX) {
+  if (name_size == 0) {
+    return GRN_SUCCESS;
+  }
+  if (name == NULL || (name_size > 0 && *name == GRN_DB_PSEUDO_COLUMN_PREFIX)) {
```

Comment:
* We don't use `== NULL`/`!= NULL`.
* `name_size > 0` is needless because we have `name_size == 0`.

```suggestion
  if (!name || *name == GRN_DB_PSEUDO_COLUMN_PREFIX) {
```

Reply:
fix: [83d4fea](https://github.com/groonga/groonga/pull/2339/commits/83d4feaaf359cd8794d4f038b636e7b70ae62210) Thank you so much.